### PR TITLE
Move key bundle types to kbfsmd package

### PR DESCRIPTION
**Pull request by akalin-keybase**
This is the first step towards moving BareRootMetadata to kbfsmd.

Also remove dependency of BareRootMetadata on cryptoPure.

Move encryptedData and friends to kbfscrypto.

### CODE REVIEW

File: kbfsmd/key_bundle.go [hunk 1]
```diff
@@ -0,0 +1,252 @@
+// Copyright 2016 Keybase Inc. All rights reserved.
```
Comment (akalin-keybase) [inline kbfsmd/key_bundle.go]: I don't know why this wasn't picked up as a rename

File: libkbfs/key_bundle.go [hunk 2]
```diff
@@ -1,261 +1,73 @@
...
-		return false
-	}
-
-	for u, dpk := range udpk {
-		if !dpk.Equals(other[u]) {
-			return false
-		}
-	}
+// TLFCryptKeyServerHalfID is a temporary alias.
+type TLFCryptKeyServerHalfID = kbfscrypto.TLFCryptKeyServerHalfID
```
Comment (akalin-keybase) [inline libkbfs/key_bundle.go]: This requires go 1.9

**Comment by akalin-keybase** [comment]
Review suggestion: review each commit one at a time, since I rebased it to be cleaner.

**Comment by akalin-keybase** [comment]
Hmm...interesting appveyor failure.

**Comment by akalin-keybase** [comment]
Looks like that's a go 1.9 thing. Will update on another PR.

**Comment by akalin-keybase** [comment]
will fix circleci separately

**Comment by akalin-keybase** [comment]
Okay, I think this is finally ready for review.

**Comment by akalin-keybase** [comment]
Ping!

File: kbfsmd/key_bundle.go (hunk 1 above)
Comment (jzila) [reply kbfsmd/key_bundle.go]: FWIW you can force it by doing `git mv foo/file bar/file`

Comment (jzila) [reply kbfsmd/key_bundle.go]: This file in particular would be valuable as a move because `crypto` has been removed from most of the methods.

**Comment by akalin-keybase** [comment]
Yeah, that's correct re. `kbfscrypto`. We may still want to keep around the `Crypto` interface in `libkbfs` (to e.g. fake out random number generation in tests) but I'd want to be able to move BareRootMetadata out of libkbfs without dragging Crypto with it.

File: kbfsmd/key_bundle.go (hunk 1 above)
Comment (akalin-keybase) [reply kbfsmd/key_bundle.go]: I tried making a separate commit with just `git mv` but that didn't work either. I think that GitHub just uses the default git heuristic for detecting renames and `key_bundle.go` changed enough to make that heuristic fail.

Comment (akalin-keybase) [reply kbfsmd/key_bundle.go]: Ah, maybe it's because there's still a file in libkbfs named `key_bundle.go`. Let me try renaming that...

Comment (akalin-keybase) [reply kbfsmd/key_bundle.go]: That worked! 🎊

**Comment by steev** [comment]
Does this change require go 1.9?  On my debian box, when attempting to build, I'm getting 

```
tmp/keybase_build_2017_10_12_191125/gopaths/arm64/src/github.com/keybase/kbfs/libkbfs/data_types.go:78: syntax error: unexpected = in type declaration
/tmp/keybase_build_2017_10_12_191125/gopaths/arm64/src/github.com/keybase/kbfs/libkbfs/data_types.go:86: syntax error: unexpected = in type declaration
/tmp/keybase_build_2017_10_12_191125/gopaths/arm64/src/github.com/keybase/kbfs/libkbfs/data_types.go:89: syntax error: unexpected = in type declaration
/tmp/keybase_build_2017_10_12_191125/gopaths/arm64/src/github.com/keybase/kbfs/libkbfs/data_types.go:92: syntax error: unexpected = in type declaration
/tmp/keybase_build_2017_10_12_191125/gopaths/arm64/src/github.com/keybase/kbfs/libkbfs/data_types.go:95: syntax error: unexpected = in type declaration
/tmp/keybase_build_2017_10_12_191125/gopaths/arm64/src/github.com/keybase/kbfs/libkbfs/data_types.go:116: syntax error: unexpected = in type declaration
```

Which is directly related to this patch - is there a missing dependency somewhere?

**Comment by jzila** [comment]
@steev: yes

**Pull request merged**